### ISSUE TITLE
RandZoom on batch

### ISSUE THREAD

**Issue by drbeh** [Project-MONAI/MONAI#2929]
**Is your feature request related to a problem? Please describe.**
`RandZoom` uses `Zoom` which uses [`torch.nn.functional.interpolate`](https://pytorch.org/docs/stable/generated/torch.nn.functional.interpolate.html) under the hood to extract randomly zoomed images. Although `torch.nn.functional.interpolate` is capable of being applied on batch x channels x [optional depth] x [optional height] x width, `Zoom` and RandZoom` don't.

Instead, they treat the channel dimension as spatial dimension. This is causing an issue since there is no error and warning if user apply it on a batched dataset.

This requires to distinguish between 3D multi-channel images vs 2D multi-channel batched images. Although an automatic logic would be hard to achieve, a simple flag can do the work.

**Comment by wyli** [comment]
the concept of minibatch is part of the model optimization using stochastic gradient descent, anything that dealing with minibatches should be in the 'monai.networks' module of monai.

The price for mixing the concepts of batch and nonbatch is not just a simple flag, it requires a full set of dimensionality input validation, and potentially reduces the usability/code readability.

**Comment by drbeh** [comment]
Hi @wyli,  isn't it a too restrictive assumption? 
What about validation and testing (inference in general)? There is no optimization there but mini-batch is required for computational efficiency.

**Comment by drbeh** [comment]
Do we have any decision here? 
Is there any technical issue that prevents us from supporting batch transform for RandZoom? 
Thanks
